### ISSUE TITLE
CA1806 does not fire for `int.TryParse(string, NumberStyles, IFormatProvider, out int)

### ISSUE THREAD

**Issue by canton7** [dotnet/roslyn-analyzers#1860]
#### Analyzer package

[Microsoft.CodeQuality.Analyzers](https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers)

#### Package Version

v2.6.2

#### Diagnostic ID

Example: [CA1806](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1806-do-not-ignore-method-results)

#### Repro steps

Call the method:

```csharp
int.TryParse(123", NumberStyles.Integer, CultureInfo.InvariantCulture, out int result)
```

#### Expected behavior

CA1806 fires, as the result return value of `int.TryParse` is not used.

#### Actual behavior

CA1806 does not fire.

The problem [is here](https://github.com/dotnet/roslyn-analyzers/blob/32d8f1e397439035f0ecb5f61a9e672225f0ecdb/src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/DoNotIgnoreMethodResults.cs#L308): the analyzer looks only fires if the second parameter is a ref/out parameter, but in this overload of TryParse the out parameter is the 4th parameter.

**Comment by canton7** [comment]
I had a cursory glance through [the TryParse methods in .NET Framework](https://docs.microsoft.com/en-us/dotnet/api/?view=netframework-4.7.2&term=tryparse): all of the TryParse methods I could find which had an `out` parameter had it has the final parameter. However, there were several `TryParse` methods which returned a bool but had no `out` parameters at all, e.g. [here](https://docs.microsoft.com/en-gb/dotnet/api/system.servicemodel.syndication.workspace.tryparseelement?view=netframework-4.7.2#System_ServiceModel_Syndication_Workspace_TryParseElement_System_Xml_XmlReader_System_String_) and [here](https://docs.microsoft.com/en-gb/dotnet/api/system.servicemodel.syndication.servicedocument.tryparseelement?view=netframework-4.7.2#System_ServiceModel_Syndication_ServiceDocument_TryParseElement_System_Xml_XmlReader_System_String_) and [here](https://docs.microsoft.com/en-gb/dotnet/api/system.servicemodel.syndication.resourcecollectioninfo.tryparseattribute?view=netframework-4.7.2#System_ServiceModel_Syndication_ResourceCollectionInfo_TryParseAttribute_System_String_System_String_System_String_System_String_) and [here](https://docs.microsoft.com/en-gb/dotnet/api/system.servicemodel.syndication.syndicationfeedformatter.tryparseattribute?view=netframework-4.7.2#System_ServiceModel_Syndication_SyndicationFeedFormatter_TryParseAttribute_System_String_System_String_System_String_System_ServiceModel_Syndication_SyndicationCategory_System_String_). This makes me wonder whether the test should be for the final parameter to be ref/out, not the second; or even whether this test should be there at all.

**Comment by mavasani** [comment]
Default behavior of the rule matches legacy FxCop, which is required for compat reasons. We can consider making this rule end user configurable to allow analyzing additional symbols.
